applications/FluidDynamicsApplication/custom_elements/data_containers/qs_vms/qs_vms_adjoint_element_data.h

```diff
@@ -0,0 +1,151 @@
+                                                            >,
+                                                        CalculationDataContainers<
+                                                            std::tuple<
+                                                                Data>,
+                                                            std::tuple<
+                                                                SubAssembly<AccelerationDerivativeContributions<0>, ElementDataContainerIndex, 0, ResidualColumnOffset>,
+                                                                SubAssembly<AccelerationDerivativeContributions<1>, ElementDataContainerIndex, 1, ResidualColumnOffset>,
+                                                                SubAssembly<AccelerationDerivativeContributions<2>, ElementDataContainerIndex, 2, ResidualColumnOffset>,
+                                                                SubAssembly<ZeroDerivatives<TNumNodes, 4>,          ElementDataContainerIndex, 3, ResidualColumnOffset>>
+                                                            >
+                                                        >::type;
+
+    using ResidualShapeDerivatives = typename std::conditional<
+                                            TDim == 2,
+                                            CalculationDataContainers<
+                                                std::tuple<
+                                                    Data>,
+                                                std::tuple<
+                                                    SubAssembly<ShapeDerivatives<0>, ElementDataContainerIndex, 0, ResidualColumnOffset>,
+                                                    SubAssembly<ShapeDerivatives<1>, ElementDataContainerIndex, 1, ResidualColumnOffset>>
+                                                >,
+                                            CalculationDataContainers<
+                                                std::tuple<
+                                                    Data>,
+                                                std::tuple<
+                                                    SubAssembly<ShapeDerivatives<0>, ElementDataContainerIndex, 0, ResidualColumnOffset>,
+                                                    SubAssembly<ShapeDerivatives<1>, ElementDataContainerIndex, 1, ResidualColumnOffset>,
+                                                    SubAssembly<ShapeDerivatives<2>, ElementDataContainerIndex, 2, ResidualColumnOffset>>
+                                                >
+                                            >::type;
+
+    ///@}
+    ///@name Static Operations
+    ///@{
+
+    static void Check(
+        const Element& rElement,
+        const ProcessInfo& rProcessInfo);
+
+    static std::vector<const Variable<double>*> GetDofVariablesList();
```

Comment:
Why you need this? DOFs are already known for each specialization as we've a template on dimensions.

Reply:
removed.